packages/xrpl/src/models/transactions/transaction.ts

```diff
@@ -307,6 +309,19 @@ export function validate(transaction: Record<string, unknown>): void {
       validateAccountSet(tx)
       break
 
+    case 'Batch':
+      validateBatch(tx)
+      // This is done here to avoid issues with dependency cycles
```

Comment:
what is done here? Which issues of the dependency cycle are you referring to?

Reply:
You can't call `validate` from within `validateBatch` on the inner transactions because that would result in a dependency cycle. So this was my way around that.